keras/callbacks.py

```diff
@@ -790,7 +790,7 @@ def set_model(self, model):
             self.sess = K.get_session()
         if self.histogram_freq and self.merged is None:
             for layer in self.model.layers:
-                for weight in layer.weights:
+                for weight in layer.trainable_weights:
```

Comment:
This is too restrictive. Rather, the conditional below should be `if self.write_grads and weight in layer.trainable_weights`